code/game/objects/items/devices/flash/core.dm

```diff
@@ -0,0 +1,235 @@
+/obj/item/device/flash
+	name = "incomplete flash assembly"
+	desc = "Used mainly by civilian sector security forces for a (mostly) harmless way of disorienting organic and synthetic beings via intense directional light."
+	icon = 'icons/obj/flash.dmi'
+	icon_state = "base"
+	item_state = "flashtool"
+	throwforce = 5
+	w_class = 2
+	throw_speed = 4
+	throw_range = 10
+	flags = CONDUCT
+	origin_tech = list(TECH_MAGNET = 2, TECH_COMBAT = 1)
+	var/cell_type = null //The type of cell it uses.
+	var/bulb_type = null //The type of bulb it uses.
+
+	//Do not change these values
+	var/strength_mul = 1 //Strength multiplier of the strength multiplier. Internal value.
+	var/obj/item/weapon/cell/power_supply //The stored power cell. Internal value.
+	var/obj/item/weapon/flash_bulb/bulb //The installed bulb. Internal value.
+
+	var/self_recharge = FALSE
+	var/use_external_power = FALSE
+	var/charge_cost = 100 //How much charge to take away per use
+
+	var/recharge_time = 4//No clue
+
+	var/DRM = FALSE //True means you can't modify it.
+
+/obj/item/device/flash/Initialize() //Stolen from energy gun code.
+	. = ..()
+	if(cell_type)
+		power_supply = new cell_type(src)
+
+	if(bulb_type)
+		bulb = new bulb_type(src)
+
+	if(bulb)
+		bulb.on = TRUE
+
+	update_icon()
+
+/obj/item/device/flash/examine(mob/user)
+	..(user)
+	var/uses_remaining = round(power_supply.charge / charge_cost)
+	to_chat(user,"Has [uses_remaining] use\s remaining.")
```

Comment:
Check the parent call's return value, you shouldn't be able to check this from across the room.